pallets/gear-builtin/src/lib.rs

```diff
@@ -212,6 +216,42 @@ pub mod pallet {
         pub fn generate_actor_id(builtin_id: u64) -> ProgramId {
             hash((SEED, builtin_id).encode().as_slice()).into()
         }
+
+        pub(crate) fn dispatch_call(
+            origin: T::AccountId,
```

Comment:
I think that could avoid misuse of a func 
```suggestion
            program: ProgramId,
```